cloudify_nsx/library/nsx_security_policy.py

```diff
@@ -60,34 +48,26 @@ def add_policy(client_session, name, description, precedence, parent,
 def add_policy_group_bind(client_session, security_policy_id,
                           security_group_id):
 
-    raw_result = client_session.read(
+    security_policy = common.nsx_read(
+        client_session, 'body',
         'securityPolicyID', uri_parameters={'ID': security_policy_id}
     )
 
-    common.check_raw_result(raw_result)
-
-    security_policy = raw_result['body']
-
-    bindings = security_policy['securityPolicy'].get(
-        'securityGroupBinding', []
+    bindings = common.nsx_struct_get_list(
+        security_policy, 'securityPolicy/securityGroupBinding'
     )
 
-    if isinstance(bindings, dict):
-        bindings = [bindings]
-
     for bind in bindings:
         if bind.get('objectId') == security_group_id:
             raise cfy_exc.NonRecoverableError(
                 "Group %s already exists in %s" % (
                     security_group_id,
-                    security_policy['securityPolicy']['name']
+                    security_policy['securityPolicy'].get('name')
```

Comment:
Maybe this should return 'unknown security policy' if the name is missing- or even something like the ID if that's possible? (I'm not sure what situation is envisaged to result in the security policy not having its name populated at this point).